hedera-node/docs/design/services/token-service/airdrops/airdrop-pending-state.md

```diff
@@ -0,0 +1,123 @@
+
+`StateDefinition.*onDisk*(*AIRDROPS_KEY*, PendingAirdropId.PROTOBUF, PendingAirdropValue.PROTOBUF, MAX__AIRDROPS);`
+
+We can have a case, where we have multiple airdrops for the same fungible token. In this case the  `PendingAirdropValue` would be updated with the new aggregated value.
+
+### Stores
+
+We need to define a special `WritableStore` for the airdrops, so that we can manipulate the content in them. This would be the `WritableAirdropStore`.
+
+It would extend `ReadableAirdropStoreImpl` that implements the following interface:
+
+```java
+public interface ReadableAirdropStore {
+	PendingAirdropValue getFungibleAirdropAmount(@NonNull final PendingAirdropId airdropId);
+	long sizeOfState();
+	void warm(@NonNull final PendingAirdropId airdropId);
+    List<PendingAirdropId> getPendingAirdropIdsBySender(final AccountId sender);
+}
+```
+
+The `ReadableAirdropStoreImpl` would keep an instance of `ReadableKVState<PendingAirdropId, PendingAirdropValue> airdropsState`.
+
+An additional method `getPendingAirdropIdsBySender` would be added to the `ReadableAirdropStore` to get all `PendingAirdropIds` for a specific sender. This would be done via filtering all `PendingAirdropIds` kept in state by the `sender` that is passed as a parameter.
+
+This would be useful for traversing all sender's airdrops when their account expires or gets deleted. Then we should iterate over all of their pending airdrops and cancel them.
+
+The `WritableAirdropStore` would add additional methods for operating over the airdrops state, so it would have the following methods:
+
+```java
+public class WritableAirdropStore {
+
+    void put(@NonNull final PendingAirdropId airdropId, @NonNull final PendingAirdropValue airdropValue) {
+        // invoked by the `TokenAirdropTransaction`
+    }
+
+    void remove(@NonNull PendingAirdropId airdropId) {
+        // invoked by the `TokenCancelAirdrop`
+    }
+}
+```
+
+### Operations
+
+1. Fetching airdrop values from the collections. This would be applicable only for fungible airdrops (for both `TokenAirdrop` and `TokenClaimAirdrop` transactions).
+   For a specific `PendingAirdropId`, we would get the corresponding `PendingAirdropValue`.
+2. Handling a new airdrop in state. We can have several cases:
+   - Only a single fungible or non-fungible airdrop for a specific token defined - we just put a new entry in the map
+   - An airdrop for an existing `PendingAirdropId` is already defined - we should replace the existing `PendingAirdropValue` for this `PendingAirdropId` with the aggregated new fungible amount
+3. Handling an airdrop `cancel` in state - we remove the `PendingAirdropId` key for the airdrop we want to cancel
+4. Handling an airdrop `claim` in state - we remove the `PendingAirdropId` key for the airdrop after we claim it and create a synthetic crypto transfer for it inside the `TokenClaimAirdropHandler`
```

Comment:
Nit: I believe we agreed that we do not want to create synthetic crypto transfers in this case.

Reply:
Removed